### PR TITLE
Feature: Dark mode and user styles

### PR DESCRIPTION
**Pull request by daniel-simpson**
Initial implementation for comment.

Addresses dark mode request: #57 along with my comment for a place to inject user-defined styles.

I'm expecting that the colors I've picked for dark mode (especially the accent colors) may need some feedback, but it's a starting point.

Thanks again for your fantastic frontend project!

### CODE REVIEW

File: public/userstyles-dark-example.css
Comment (daniel-simpson) [inline public/userstyles-dark-example.css:1]: Happy to remove this whole file, as the styles I've added here are also [here](https://github.com/karlomikus/vue-salt-rim/pull/63/files#diff-228cf18ff605a4b69cd51f90a284390689ffe95de8261795304d9edd23050dc5R57)

File: public/userstyles.css
Comment (daniel-simpson) [inline public/userstyles.css:1]: Intentionally left blank.

Ideally, we could leave this here, and provide a place for users to inject their own styles using docker file volumes, e.g.:

```yaml
  salt-rim:
    image: barassistant/salt-rim:latest
    depends_on:
      - bar-assistant-api
    volumes:
      - ./my-custom-userstyles.css:/var/www/html/userstyles.css
    ...
```

Allowing users to style their Salt Rim instance more like their home bars 💌

File: src/assets/base.css [hunk 1]
```diff
@@ -45,24 +45,89 @@
...
+        
+        --clr-empty-state: var(--clr-gray-700);
+        --clr-empty-state-svg: var(--clr-gray-200);
+    
+        --clr-accent-blue: #0f3c54;
+        --clr-accent-green: #0a4638;
+        --clr-accent-yellow: #d1ba1d;
+        --clr-accent-red: #991922; 
+        --clr-accent-purple: #66158f;
+        --clr-accent-orange: #df7c27;
```
Comment (daniel-simpson) [inline src/assets/base.css:109]: These accent colors are a starting point for discussion, obviously they are nothing like the original palette.  Happy to modify to whatever you'd prefer!

File: vite.config.js [hunk 2]
```diff
@@ -23,12 +23,28 @@ const manifest = {
...
 
+function externalCSSPlugin() {
+    return {
+        name: 'external-userstyles',
+        transformIndexHtml: {
+            enforce: 'post',
+            transform() {
+                return [{
+                    tag: "link",
+                    attrs: {"rel": "stylesheet", "type": "text/css", "href": "/userstyles.css"},
```
Comment (daniel-simpson) [inline vite.config.js:34]: Injects the empty `userstyles.css` to the end of the HTML head element, allowing users to inject their own styles (without having to `!important;` everything)

**Comment by daniel-simpson** [comment]
A few example screenshots:

image

image

**Comment by karlomikus** [comment]
I'm closing this in favor of current develop branch changes.

**Pull request closed without merging**